subcommand/health-sync/command_test.go

```diff
@@ -0,0 +1,255 @@
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+)
+
+func TestRunWithoutContainerNames(t *testing.T) {
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI: ui,
+	}
+	code := cmd.Run(nil)
+	require.Equal(t, 0, code)
+}
+
+func TestEcsHealthToConsulHealth(t *testing.T) {
+	require.Equal(t, api.HealthPassing, ecsHealthToConsulHealth("HEALTHY"))
+	require.Equal(t, api.HealthCritical, ecsHealthToConsulHealth("UNHEALTHY"))
+	require.Equal(t, api.HealthCritical, ecsHealthToConsulHealth("UNKNOWN"))
+	require.Equal(t, api.HealthCritical, ecsHealthToConsulHealth(""))
+}
+
+type minimalContainerMetadata struct {
+	name      string
+	status    string
+	essential bool
+}
+
+var (
+	serviceName = "Family"
+	taskID      = "TaskId"
+	taskARN     = fmt.Sprintf("asdf/%s", taskID)
+	serviceID   = fmt.Sprintf("%s-%s", serviceName, taskID)
+)
+
+func metadataResponse(containerMetadatas []minimalContainerMetadata) string {
+	var taskMetaContainers []awsutil.ECSTaskMetaContainer
+	for _, containerMetadata := range containerMetadatas {
+		metaContainer := awsutil.ECSTaskMetaContainer{
+			Name: containerMetadata.name,
+		}
+		metaContainer.Health.Status = containerMetadata.status
+		taskMetaContainers = append(taskMetaContainers, metaContainer)
+	}
+
+	taskMeta := awsutil.ECSTaskMeta{
+		Family:     serviceName,
+		TaskARN:    taskARN,
+		Containers: taskMetaContainers,
+	}
+	json, _ := json.Marshal(taskMeta)
+	return string(json)
+}
+
+func setupServiceAndChecks(t *testing.T, consulClient *api.Client, containers []minimalContainerMetadata) {
+	checks := api.AgentServiceChecks{}
+	for _, container := range containers {
+		if !container.essential {
+			continue
+		}
+
+		checks = append(checks, &api.AgentServiceCheck{
+			CheckID: makeCheckID(serviceName, container.name),
+			TTL:     "10000h",
+		})
+	}
+
+	err := consulClient.Agent().ServiceRegister(&api.AgentServiceRegistration{
+		ID:   serviceID,
+		Name: serviceName,
+		Port: 1000,
+		Meta: map[string]string{
+			"task-id":  taskID,
+			"task-arn": taskARN,
+			"source":   "consul-ecs",
+		},
+		Checks: checks,
+	})
+	require.NoError(t, err)
+}
+
+func assertHealthChecks(t *testing.T, consulClient *api.Client, containers []minimalContainerMetadata) {
+	filter := fmt.Sprintf(`ServiceID == "%s"`, serviceID)
+	checks, err := consulClient.Agent().ChecksWithFilter(filter)
+	require.NoError(t, err)
+	for _, container := range containers {
+		checkID := makeCheckID(serviceName, container.name)
+		check, ok := checks[checkID]
+
+		if !container.essential {
+			require.False(t, ok)
+			continue
+		}
+
+		require.True(t, ok)
+		require.Equal(t, ecsHealthToConsulHealth(container.status), check.Status)
+	}
+}
+
+var cases map[string][]minimalContainerMetadata = map[string][]minimalContainerMetadata{
+	"one essential healthy container": {{
```

Comment:
This was driving me crazy - I think `go fmt` just needs a nudge.

If you stick a newline like this:

```
    "one essential healthy container": {
    {
```

And then run `go fmt`, it will format like the following:

```
    "one essential healthy container": {
        {
            name:      "container1",
            status:    "HEALTHY",
            essential: true,
        },
    },
```

Reply:
Nice find. `go fmt` isn't advanced enough to deal with my attempt to format Go like Clojure 😆